Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift

```diff
@@ -81,4 +81,12 @@ extension StorageCategory: StorageCategoryBehavior {
                      resultListener: StorageListOperation.ResultListener?) -> StorageListOperation {
         return plugin.list(options: options, resultListener: resultListener)
     }
+
+    // MARK: - Async API -
+
+    @discardableResult
+    public func getURL(key: String,
```

Comment:
1. Are we keeping the call back apis?
2. Let make the Options take a default nil  `options: StorageGetURLOperation.Request.Options? = nil`

Reply:
the listener API is unchanged for this PR. once the async functions are merged to dev-preview the listener API can be removed. this will make for a more focused PR for each change.